app/shop.py

```diff
@@ -0,0 +1,38 @@
+from __future__ import annotations
+
+from app.customer import Customer
+
+
+class Shop:
+    def __init__(self, name: str, location: list, products: dict) -> None:
+        self._name = name
+        self._location = location
+        self._products = products
+
+    @classmethod
+    def create_shop(cls, shop_info: dict) -> Shop:
+        return cls(
+            shop_info["name"],
+            shop_info["location"],
+            shop_info["products"]
+        )
+
+    @property
+    def name(self) -> str:
+        return self._name
+
+    @property
+    def location(self) -> list:
+        return self._location
+
+    @property
+    def products(self) -> dict:
+        return self._products
+
+    def calculate_total_product_cost(self, customer: Customer) -> int:
+        products_total_cost = 0
+        for product in self._products:
+            products_total_cost += (self._products[product]
+                                    * customer.products[product])

```

Comment:
The `calculate_total_product_cost` method assumes that the customer always wants to buy products available in the shop. Consider adding a check to ensure that the product exists in the customer's product list before attempting to access it, to avoid potential `KeyError` exceptions.